lib/buffer.js

```diff
@@ -326,13 +326,34 @@ Object.defineProperty(Buffer.prototype, 'offset', {
 function slowToString(encoding, start, end) {
   var loweredCase = false;
 
-  start = start >>> 0;
-  end = end === undefined || end === Infinity ? this.length : end >>> 0;
+  if (start >= this.length || end <= 0 || Number.isNaN(end))
```

Comment:
I forget. What was the rational behind checking for `NaN` explicitly while allowing other values that would coerce to `NaN` get through?